docs/content/development/modules/create.mdx

```diff
@@ -0,0 +1,321 @@
+    moduleDeclaration: InternalModuleDeclaration
+  ): Promise<void>
+  revertMigration?(
+    options: LoaderOptions,
+    moduleDeclaration: InternalModuleDeclaration
+  ): Promise<void>
+}
+```
+
+:::note
+
+All property types such as `ModuleLoaderFunction` can be loaded from the `@medusajs/modules-sdk` package.
+
+:::
+
+Where:
+
+- `service`: This is the only required property to be exported. It should be the main service your module exposes.
+- `loaders`: (optional) an array of functions used to perform an action while loading the module. For example, you can log a message that the module has been loaded.
+- `migrations`: (optional) an array of paths to migration files that should run when the `migration` command is used with Medusa's CLI.
+- `models`: (optional) an array of entities that your module creates.
+- `runMigrations`: (optional) a function that can be used to define migrations to run when the `migration run` command is used with Medusa's CLI. The migrations will only run if they haven't already.
+- `revertMigration`: (optional) a function can be used to define how migrations should be reverted when the `migration revert` command is used with Medusa's CLI.
+
+Here's an example implementation of `src/index.ts` from Medusa's Redis Cache module:
+
+```ts title=src/index.ts
+import { ModuleExports } from "@medusajs/modules-sdk"
+
+import Loader from "./loaders"
+import { RedisCacheService } from "./services"
+
+const service = RedisCacheService
+const loaders = [Loader]
+
+const moduleDefinition: ModuleExports = {
+  service,
+  loaders,
+}
+
+export default moduleDefinition
+```
+
+---
+
+## Step 6: Test your Module
+
+This step explains how you can test your module in a locally installed [Medusa backend](../backend/install.mdx) after or while implementing your custom functionality.
+
+Run the following command in the root of your module directory:
+
+```bash npm2yarn
+npm run build
+```
+
+Then, register your package locally to use it in other NPM projects with the following command:
+
+```bash npm2yarn
+npm link
+```
+
+Next, in your Medusa backend, run the following command to install your module:
+
+```bash npm2yarn
+npm link module-name
+```
+
+Where `module-name` is the name of your module as defined in the `package.json` of your module.
+
+Finally, add your module to `medusa-config.js`:
+
+```js title=medusa-config.js
+module.exports = {
+  // ...
+  modules: { 
+    // ...
+    moduleType: {
+      resolve: "<module_name>", 
+      options: {
+        // options if necessary
+      },
+    },
+  },
+}
+```
+
+The way you add your module depends on its type and what options it requires, if any. Note that in the above code example:
+
+- `moduleType` is the type of your module. For example, if your module is a cache module, it should be changed to `cacheService`.
+- `<module_name>` is the name of your module as defined in the `package.json` of your module.
+- `options` should hold any options of your module, if necessary.
+
```

Comment:
Maybe we could expand this and add the `resources` property that is inside `moduleType`. at the same level as the `resolve`

resouces can be:

`shared` (this is the default) - All the Awilix container is shared with the module. Including the database connection. The db connection can be skipped if the module is sharing resources with Medusa's Core.

`isolated` - The module will receive an empty Awilix container, and only dependencies will be registered there. These dependencies are defined at the Module's definition and cannot be changed. The module will be responsible for managing it's own database connection, and other resources.